plutus-contract/src/Language/Plutus/Contract/StateMachine.hs

```diff
@@ -333,7 +336,8 @@ mkStep ::
     -> Contract schema e (StateMachineTransition state input)
 mkStep client@StateMachineClient{scInstance} input = do
     let StateMachineInstance{stateMachine=StateMachine{smTransition}, validatorInstance} = scInstance
-    (onChainState, utxo) <- getOnChainState client
+    maybeState <- getOnChainState client
+    (onChainState, utxo) <- maybe (throwing_ _NoOnChainState) pure maybeState
```

Comment:
Maybe we should propagate it upwards here too... not sure.

Reply:
Relevant to Jann's open PR.

Reply:
I ended up removing the `NoOnChainState` failure in https://github.com/input-output-hk/plutus/pull/2511. There is now a type

```haskell
data InvalidTransition = InvalidTransition
        { tfState :: Maybe (State s)
        , tfInput :: i
        }
```

where `tfState = Nothing` is used for the case that no on-chain state was found.